### DISCUSSION TITLE
Setting up SSH

### DISCUSSION THREAD

**Discussion by hphillingham** [AlexGustafsson/homebridge-wol#141] [Questions] [answered]
Hey, I'm having some trouble getting my Mac to respond to the SSH commands. 
Homebridge is installed and running on the same Mac, under a <macname>.local address. I can't seem to SSH into my machine to get any of the default commands to work, as I'm given the error "Host Key verification failed"
I'm completely new to SSH so a really dumbed down answer would be a massive help - Thanks!

**Comment by AlexGustafsson** [comment, marked as answer]
On the machine running Homebridge, SSH into your Mac manually via a terminal. `Host Key verification failed` tells us that your client does not trust the target machine, likely indicating that you haven't entrusted the host yet.

**Reply by hphillingham** [reply]
I'm running Homebridge on the same Mac as the one I'm trying to SSH into. I don't know if I can/how to SSH between the part running Homebridge and the terminal. If not, no worries
I'm also getting a "Permission denied, disconnected due to too many authentication errors" error. Do I need to include a password in my SSH command? Again totally new to this, if its a lost cause then  no worries

**Reply by AlexGustafsson** [reply]
If you're running Homebridge on the same device, there's no need for SSH. You may simply use the Mac's IP address (or better yet, the loopback address `127.0.0.1`) and get rid of any ping command etc. you may have. That way homebridge-wol will simply ping the device (itself). Yet another way is to remove the IP address and use the ping command `true`. In that case the program called `true` will be executed instead of pinging - always returning `true` (exiting successfully). The result will be that the Mac will show up as online for as long as it's on. When it's non-reachable, neither is Homebridge itself, meaning you won't be able to read the correct status anyway.

The error you're seeing is indeed related to authentication, but there are many different reasons as to why. One of which is that you may have to enter your password, which can only be done "interactively" - that is, you must run the command via an interactive command prompt and not as part of another "automated" program such as homebridge-wol.

There's never a lost cause, only varying degrees of potential of learning new things! :)

**Reply by hphillingham** [reply]
Ahh this worked an absolute treat, thank you !
